File: app/src/main/java/id/lima/baseapp/ui/splash/SplashContract.java
package id.lima.baseapp.ui.splash;

import id.lima.baseapp.base.BasePresenter;
import id.lima.baseapp.base.BaseView;

/**
 * Created with love by muhwid on 27/10/18
 **/

public interface SplashContract {
    interface View extends BaseView<Presenter> {
        void showLoginProgress();

        void showLoginSuccess();

        void showLoginError(String message);
    }

    interface Presenter extends BasePresenter {
        void onLogin(String username, String password);
    }
}

File: app/src/main/java/id/lima/moviecatalog/ui/splash/SplashContract.kt
package id.lima.moviecatalog.ui.splash

import android.content.Context
import android.content.Intent
import id.lima.moviecatalog.base.BasePresenter
import id.lima.moviecatalog.base.BaseView

/**
 * Created with love by muhwid on 17/04/20.
 * email muhwid29@gmail.com
 */

interface SplashContract {
    interface View : BaseView<Presenter?> {

    }

    interface Presenter : BasePresenter {
        fun moveToNext(intent: Intent)
    }
}